pyomo/contrib/mindtpy/feasibility_pump_new.py

```diff
@@ -0,0 +1,419 @@
+                feas_main_results.solution.status is SolutionStatus.feasible):
+            config.logger.warning('MILP solver reported feasible solution of FP-MIP, '
+                                'but not guaranteed to be optimal.')
+            return False
+        else:
+            config.logger.warning('Unexpected result of FP-MIP')
+            return True
+
+
+    def fp_loop(self, config):
+        """Feasibility pump loop.
+
+        This is the outermost function for the algorithms in this package; this function
+        controls the progression of solving the model.
+
+        Parameters
+        ----------
+        solve_data : MindtPySolveData
+            Data container that holds solve-instance data.
+        config : ConfigBlock
+            The specific configurations for MindtPy.
+
+        Raises
+        ------
+        ValueError
+            MindtPy unable to handle the termination condition of the FP-NLP subproblem.
+        """
+        while self.fp_iter < config.fp_iteration_limit:
+
+            self.mip_subiter = 0
+            # solve MILP main problem
+            feas_main, feas_main_results = self.solve_main(config, fp=True)
+            fp_should_terminate = self.handle_fp_main_tc(feas_main_results, config)
+            if fp_should_terminate:
+                break
+
+            # Solve NLP subproblem
+            # The constraint linearization happens in the handlers
+            fp_nlp, fp_nlp_result = self.solve_fp_subproblem(config)
+
+            if fp_nlp_result.solver.termination_condition in {tc.optimal, tc.locallyOptimal, tc.feasible}:
+                config.logger.info(self.log_formatter.format(
+                    self.fp_iter, 'FP-NLP', value(
+                        fp_nlp.MindtPy_utils.fp_nlp_obj),
+                    self.primal_bound, self.dual_bound, self.rel_gap,
+                    get_main_elapsed_time(self.timing)))
+                self.handle_fp_subproblem_optimal(fp_nlp, config)
+            elif fp_nlp_result.solver.termination_condition in {tc.infeasible, tc.noSolution}:
+                config.logger.error('Feasibility pump NLP subproblem infeasible')
+                self.should_terminate = True
+                self.results.solver.status = SolverStatus.error
+                return
+            elif fp_nlp_result.solver.termination_condition is tc.maxIterations:
+                config.logger.error(
+                    'Feasibility pump NLP subproblem failed to converge within iteration limit.')
+                self.should_terminate = True
+                self.results.solver.status = SolverStatus.error
+                return
+            else:
+                raise ValueError(
+                    'MindtPy unable to handle NLP subproblem termination '
+                    'condition of {}'.format(fp_nlp_result.solver.termination_condition))
+            # Call the NLP post-solve callback
+            # TODO fix bug
```

Comment:
What do you mean with this bug?

Reply:
I don't remember. This might be temporary comment, which I have already resolved.